### PR TITLE
fixes arm64 building and X11 conflicts

### LINKED ISSUE #208: ARM64 Linux Build errors
One such error during compilation of Zelda64Recomp

```
[17041/17436] Building CXX object rt64/CMakeFiles/rt64.dir/src/hle/rt64_application_window.cpp.o
FAILED: rt64/CMakeFiles/rt64.dir/src/hle/rt64_application_window.cpp.o 
/usr/bin/clang++-15 -DFFX_GCC -DHLSL_CPU -DIMGUI_IMPL_VULKAN_NO_PROTOTYPES -I/home/gman/Zelda64Recomp/lib/rt64/src -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/D3D12MemoryAllocator/include -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/D3D12MemoryAllocator/src -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/imgui -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/hlslpp/include -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/dxc/inc -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/Vulkan-Headers/include -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/VulkanMemoryAllocator/include -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/mupen64plus-core/src/api -I/home/gman/Zelda64Recomp/lib/rt64/src/contrib/nativefiledialog-extended/src/include -I/home/gman/Zelda64Recomp/cmake-build/src -I/home/gman/Zelda64Recomp/cmake-build/rt64/src -isystem /usr/include/SDL2 -Wno-everything -Wall -Wextra -O3 -DNDEBUG -std=c++17 -fvisibility=hidden -Wno-nullability-completeness -MD -MT rt64/CMakeFiles/rt64.dir/src/hle/rt64_application_window.cpp.o -MF rt64/CMakeFiles/rt64.dir/src/hle/rt64_application_window.cpp.o.d -o rt64/CMakeFiles/rt64.dir/src/hle/rt64_application_window.cpp.o -c /home/gman/Zelda64Recomp/lib/rt64/src/hle/rt64_application_window.cpp
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/hle/rt64_application_window.cpp:18:
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/common/rt64_common.h:12:
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/common/rt64_hlslpp.h:7:
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/contrib/hlslpp/include/hlsl++.h:3:
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/contrib/hlslpp/include/hlsl++_common.h:3:
In file included from /home/gman/Zelda64Recomp/lib/rt64/src/contrib/hlslpp/include/hlsl++_config.h:169:
/home/gman/Zelda64Recomp/lib/rt64/src/contrib/hlslpp/include/platforms/hlsl++_neon.h:155:24: error: expected unqualified-id
        static const uint32_t ControlMask[4] = { 0x03020100, 0x07060504, 0x0B0A0908, 0x0F0E0D0C }; // Swizzle x, swizzle y, swizzle z, swizzle w
                              ^
/usr/include/X11/X.h:223:23: note: expanded from macro 'ControlMask'
#define ControlMask             (1<<2)
                                 ^
```

### PR DESCRIPTION
**Pull request by theofficialgman**
closes https://github.com/Zelda64Recomp/Zelda64Recomp/issues/208 and the issues linked within

ARM64 CI can come later if that is OK.

### CODE REVIEW

**Comment by Mr-Wiseguy** [comment]
The current CI is pending a fix after the repo was migrated to an org, once that's done I'll rerun the CI and review this PR.

**Comment by dcvz** [comment]
@theofficialgman would you mind making an empty commit? The repository org switch requires a new commit for secrets to take.

**Comment by dcvz** [comment]
Turns out it’s an issue with forks and secrets, we’re working on a fix

**Comment by dcvz** [comment]
@theofficialgman would you mind rebasing against `dev` so it can pull our updates to the CI?

**Comment by theofficialgman** [comment]
rebased and waiting for approval

**Comment by Mr-Wiseguy** [comment]
Looks like the CI failed as -march=nehalem is still needed for building on x86_64 to use the SIMD intrinsics. You'll need to reenable that conditionally.

**Comment by theofficialgman** [comment]
So funny thing about https://github.com/Zelda64Recomp/Zelda64Recomp/pull/214/commits/50644954d7b6a0928cf1b42724348b62905f51c8
I did not need it on my first system because I had [box64](https://github.com/ptitSeb/box64) installed and the x86_64 binary ran through it without me catching that happening during the build process

File: CMakeLists.txt [hunk 1]
```diff
@@ -181,11 +181,18 @@ target_include_directories(Zelda64Recompiled PRIVATE
...
-    -fms-extensions
-)
+if(CMAKE_SIZEOF_VOID_P EQUAL 8 AND CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|AMD64")
+    target_compile_options(Zelda64Recompiled PRIVATE
+        -march=nehalem
+        -fno-strict-aliasing
+        -fms-extensions
+    )
+else()
+    target_compile_options(Zelda64Recompiled PRIVATE
```
Comment (dcvz) [inline CMakeLists.txt:192]: What device are you building on? Is NEON automatically enabled without compiler flags?
https://github.com/DLTcollab/sse2neon?tab=readme-ov-file#usage

Comment (theofficialgman) [reply CMakeLists.txt:192]: All ARMv8 processors have SIMD. So yes, ofc it's enabled by default. <https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html>

**Review by dcvz — APPROVED** [review]
LGTM 👍 Thanks for the contribution @theofficialgman! I'll let @Mr-Wiseguy sign off on the inclusion of `sse2neon` header before I merge.

**Comment by dcvz** [comment]
@theofficialgman actually let's include sse2neon along with the other 3rd party libs in `lib`. As long as you add it to the PUBLIC includes in the CMakeLists it should be no issue for it to be imported in `rsp_vu`.

**Comment by dcvz** [comment]
Thanks again for the contribution @theofficialgman !

**Pull request merged**